modules/backendscheduler/backendscheduler.go

```diff
@@ -219,6 +219,9 @@ func (s *BackendScheduler) Next(ctx context.Context, req *tempopb.NextJobRequest
 		return resp, nil
 	}
 
+	timeoutCtx, cancel := context.WithTimeout(ctx, s.cfg.JobTimeout)
```

Comment:
this is a timeout imposed by the server. should the caller instead set a timeout and bail after N seconds when it has decided to give up? we have timeouts like this in our queriers and i've always wondered if it should be set caller side. fine either way

Reply:
I think I want to watch it with this on the server.  We can set a timeout on the grpc caller if it ends up being an issue.